vms/platformvm/txs/fee/calculator.go

```diff
@@ -13,24 +13,26 @@ import (
 
 var _ txs.Visitor = (*calculator)(nil)
 
-func NewStaticCalculator(config StaticConfig, upgradeTimes upgrade.Config) *Calculator {
+func NewStaticCalculator(config StaticConfig, upgradeTimes upgrade.Config, chainTime time.Time) *Calculator {
 	return &Calculator{
 		config:       config,
 		upgradeTimes: upgradeTimes,
+		time:         chainTime,
 	}
 }
 
 type Calculator struct {
 	config       StaticConfig
 	upgradeTimes upgrade.Config
+	time         time.Time
 }
 
 // [CalculateFee] returns the minimal fee needed to accept [tx], at chain time [time]
-func (c *Calculator) CalculateFee(tx txs.UnsignedTx, time time.Time) uint64 {
```

Comment:
Why are we moving this parameter into the constructor? I think if we keep this parameter here we can reuse the same calculator instance for all blocks.

Reply:
Apologies for the delayed answer. I cleaned up the PRs following this one, following all of your suggestions but this one.
You're right, `staticCalculator` could be instantiated once and reused per each block if we move `chainTime` among `CalculateFee` parameters, as it is now (even if `chainTime` does not change across block transactions).
The problem is that for `dynamicCalculator` (soon to come) is not that simple to be made reusable for all blocks.
`DynamicCalculator` has a `commonfee.Calculator` core that needs to be update block by block. `commonfee.Calculator` tracks `gasPrice`, `gasCap` and the gas consumed by each block, which need to be updated/reset block by block.
So to instantiate DynamicCalculator once for each block I see two alternatives:

- Add `commonfee.Calculator` as `CalculateFee` parameter, along with `chainTime`.
- Instantiate `fee.Calculator` once, but introduce an `Update` method that takes care of updating/resetting all block by block.

I don't like the first alternative and I find the second alternative not much different that what we have now.